packages/protocol/contracts/governance/Validators.sol

```diff
@@ -1131,4 +1132,30 @@ contract Validators is
       }
     }
   }
+
+  /**
+  * @notice Returns the group that `account` was a member of during `epochNumber`.
+  * @param account The account whose group membership should be returned.
+  * @param epochNumber The epoch number we are querying this account's membership at.
+  * @param index The index into the validator's history struct for their history at `epochNumber`.
+  * @return The group that `account` was a member of during `epochNumber`.
+  */
+  function groupMembershipInEpoch(address account, uint256 epochNumber, uint256 index)
+    external
+    view
+    onlyRegisteredContracts(canForceDeaffiliation)
+    returns (address)
+  {
+    require(isValidator(account) && epochNumber <= getEpochNumber());
+    MembershipHistory storage history = validators[account].membershipHistory;
+    if (
+      history.entries[index].epochNumber == epochNumber ||
+      (history.entries[index].epochNumber < epochNumber &&
+        (history.entries[index.add(1)].epochNumber > epochNumber ||
+          history.entries[index.add(1)].epochNumber == 0))
```

Comment:
I don't think this is strictly correct.

In epoch 3, I join group A (entry index N is A)
In epoch 4, I leave group A (entry index N+1 is 0)
in epoch 5, I join group B (entry index N+2 is B)

groupMembershipInEpoch(me, 5, N) == A 😢

Reply:
Please add a test case that would have caught this

Reply:
Addressed, please review again